### PR TITLE
Add router module to Vega upgrade

### PR DESCRIPTION
**Pull request by jackzampolin**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -99,6 +99,10 @@ import (
 	dbm "github.com/tendermint/tm-db"
 
 	gaiaappparams "github.com/cosmos/gaia/v6/app/params"
+	router "github.com/strangelove-ventures/packet-forward-middleware/router"
```
Comment (yaruwangway) [inline app/app.go:102]: no need to write "router" if the alias is the same name as the package

**Review by yaruwangway — COMMENTED** [review]
Hi Jack, it is great that we can include router module to gaia vega-rc2 release!
Please confirm if the following code change need to be added:
- add `routertypes.ModuleName` in `app.mm.SetOrderBeginBlockers` and `app.mm.SetOrderInitGenesis`
-  add `fromVM[routertypes.ModuleName] = 0` if the router module do not want to skip init genesis. like this:https://github.com/cosmos/gaia/blob/06d6f8f29d989da7a5275e2a664d7ca7bf1992d8/app/app.go#L565
- add the router module store in the store upgrade. like here: https://github.com/cosmos/gaia/blob/06d6f8f29d989da7a5275e2a664d7ca7bf1992d8/app/app.go#L579

**Review by alexanderbez — CHANGES_REQUESTED** [review]
This needs to go into `main` then backported into the release branch. Otherwise, LGTM.

**Comment by jackzampolin** [comment]
Ok so this is good, but I just want to make sure to test the e2e with this build. Should have this done by eow

**Comment by jackzampolin** [comment]
Will rebase against `main`

**Comment by okwme** [comment]
this work was cherry-picked by @yaruwangway and added to the `vega-rc2` branch as part of https://github.com/cosmos/gaia/pull/979. Going to close this PR unless there's another reason to keep it open.

**Pull request closed without merging**